internal/handlers/common/filter.go

```diff
@@ -452,3 +460,67 @@ func filterFieldExprBitsAnySet(fieldValue, maskValue any) (bool, error) {
 
 	return true, nil
 }
+
+// filterFieldMod handles {field: {$mod: [divisor, remainder]}} filter.
+func filterFieldMod(fieldValue, exprValue any) (bool, error) {
+	field, err := toInt64(fieldValue)
+	if err != nil {
+		return false, nil
+	}
+
+	arr := exprValue.(*types.Array)
+	if arr.Len() < 2 {
+		return false, NewErrorMsg(ErrBadValue, `malformed mod, not enough elements`)
+	}
+	if arr.Len() > 2 {
+		return false, NewErrorMsg(ErrBadValue, `malformed mod, too many elements`)
+	}
+
+	div, err := toInt64(must.NotFail(arr.Get(0)))
+	if err != nil {
+		switch err {
```

Comment:
We could move this switch to `internal/handlers/common/error.go` with name like `formatFieldModError`. It would make this function a bit smaller.

Reply:
But that function will not be shared, so why extract it?

Reply:
I get your point, we could leave it as is.
My point is that this
```go
if err != nil {
	return false, formatFieldModError(err)
}
```
is easier to read rather then
```go
if err != nil {
	switch err {
	case errUnexpectedType:
		return false, NewErrorMsg(ErrBadValue, `malformed mod, divisor not a number`)
	case errNotWholeNumber:
		return false, NewErrorMsg(ErrBadValue, `malformed mod, divisor value is invalid :: caused by :: `+
			`Unable to coerce NaN/Inf to integral type`)
	default:
		return false, err
	}
}
```

Reply:
Sure. Because that code translates `toInt64` errors to protocol errors. If we move both functions inside, there will be no error translation code at all. The resulting code will be both shorter and easier to read because the reader will not have a need to jump to the `toInt64` definition to see what errors it returns.